build/RuntimeStore.targets

```diff
@@ -0,0 +1,176 @@
+    <Exec Command="dotnet msbuild /t:&quot;Restore;Rebuild;CollectDeps&quot; $(_TemplatesDirectory)HostingStartup/HostingStartup.csproj /p:&quot;DepsOutputPath=$(_DepsOutputDirectory);HostingStartupPackageName=%(HostingStartupArtifacts.PackageId);HostingStartupPackageVersion=%(HostingStartupArtifacts.Version);RuntimeFrameworkVersion=$(SharedFrameworkVersion)&quot;"/>
+
+    <ItemGroup>
+      <DepsFiles Include="$(_DepsOutputDirectory)**\*.deps.json" />
+    </ItemGroup>
+
+    <RepoTasks.TrimDeps DepsFiles="@(DepsFiles)" />
+  </Target>
+
+  <Target Name="PackRuntimeStore" DependsOnTargets="_ResolveRuntimeStoreRID">
+    <PropertyGroup>
+      <OutputZip>$(ArtifactsDir)aspnetcore-store-$(VersionPrefix)-$(VersionSuffix)-$(RID).zip</OutputZip>
+      <OutputSymbolZip>$(ArtifactsDir)aspnetcore-symbols-$(VersionPrefix)-$(VersionSuffix)-$(RID).zip</OutputSymbolZip>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <RuntimeStoreManifestFiles Include="$(_RuntimeStoreOutputDirectory)**\artifact.xml" />
+      <RuntimeStoreFiles Include="$(_RuntimeStoreOutputDirectory)**\*" Exclude="$(_RuntimeStoreOutputDirectory)**\artifact.xml;$(_RuntimeStoreOutputDirectory)symbols\**\*" />
+      <RuntimeStoreSymbolFiles Include="$(_RuntimeStoreOutputDirectory)symbols\**\*" />
+      <DepsFiles Include="$(_DepsOutputDirectory)**\*" />
+      <ExistingStoreManifests Include="$(_ExistingManifestsDirectory)*.xml" />
+    </ItemGroup>
+
+    <!-- Compose new store and symbols -->
+    <RepoTasks.ComposeNewStore
+      ExistingManifests="@(ExistingStoreManifests)"
+      NewManifests="@(RuntimeStoreManifestFiles)"
+      RuntimeStoreFiles="@(RuntimeStoreFiles)"
+      RuntimeStoreSymbolFiles="@(RuntimeStoreSymbolFiles)"
+      ManifestDestination="$(ArtifactsDir)aspnetcore-store-$(VersionPrefix)-$(VersionSuffix)-$(RID).xml"
+      StoreDestination="$(_StoreZipDirectory)store\"
+      SymbolsDestination="$(_SymbolsZipDirectory)"/>
+
+    <!-- Insert deps files -->
+    <Copy SourceFiles="@(DepsFiles)" DestinationFolder="$(_StoreZipDirectory)additionalDeps\%(RecursiveDir)" />
+
+    <ItemGroup>
+      <OutputZipFiles Include="$(_StoreZipDirectory)**\*" />
+      <OutputSymbolZipFiles Include="$(_SymbolsZipDirectory)**\*" />
+    </ItemGroup>
+
+    <!-- Create zips -->
+    <ZipArchive File="$(OutputZip)" SourceFiles="@(OutputZipFiles)" WorkingDirectory="$(_StoreZipDirectory)" Overwrite="true"/>
+    <ZipArchive File="$(OutputSymbolZip)" SourceFiles="@(OutputSymbolZipFiles)" WorkingDirectory="$(_SymbolsZipDirectory)" Overwrite="true" Condition="'@(OutputSymbolZipFiles)' != ''"/>
+  </Target>
+
+  <Target Name="CreateCommonManifest" DependsOnTargets="_PrepareRuntimeStoreBuildAssets">
+    <PropertyGroup>
+      <CommonManifestFileName>aspnetcore-store-$(VersionPrefix)-$(VersionSuffix)-common.xml</CommonManifestFileName>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <PackagesToTrim Include="runtime.win-arm64.runtime.native.system.data.sqlclient.sni" />
```

Comment:
Can you add a comment about why this is excluded? Also, is this temporary or a permanent workaround? If temporary, can we put this somewhere more visible so we remember to undo in the future?

Reply:
Permanent, so I'll add a comment. The reason is that the sqlclient package brings in this dependency for non-standalone apps. However since we do not build a runtime store for `win-arm64` this package isn't included in our manifest that dictates which packages to trim. However we know for sure that this package will be included in the runtime so we should trim it from the output.